backend/app/utils/Ocr.scala

```diff
@@ -102,15 +106,36 @@ object Ocr extends Logging {
 
   // TODO MRB: allow OcrMyPdf to read DPI if set in metadata
   // OCRmyPDF is a wrapper for Tesseract that we use to overlay the OCR as a text layer in the resulting PDF
-  def invokeOcrMyPdf(lang: String, inputFilePath: Path, dpi: Option[Int], stderr: OcrStderrLogger, tmpDir: Path): Path = {
+  def invokeOcrMyPdf(lang: String, inputFilePath: Path, dpi: Option[Int], stderr: OcrStderrLogger, tmpDir: Path, numberOfPages: Option[Int]): Path = {
     val tempFile = tmpDir.resolve(s"${inputFilePath.getFileName}.ocr.pdf")
     val stdout = mutable.Buffer.empty[String]
+    val defaultTimeoutInMinutes = 300
+
+
+    // Timeout is added because ocrMyPdf may get stuck processing a file
+    // and we don't want to block the worker forever
+    def runProcessWithTimeout(processBuilder: ProcessBuilder): Int = {
+      val process = processBuilder.run(ProcessLogger(stdout.append(_), stderr.append))
+      val future = Future(blocking(process.exitValue()))
+      // timeout duration gives at most 1 minute to ocr every page
+      // default 300 minutes is used in case the number of pages failed to
+      // get retrieved from the file and the value was None at this stage
+      val timeoutDuration = numberOfPages.getOrElse(defaultTimeoutInMinutes)
```

Comment:
Pedantic but if you're working with untyped durations it can be helpful to have the unit in the name, e.g. `timeoutMinutes`

You could also define a constant `TIMEOUT_DURATION_PER_PAGE = 1`, which might make the code scan more easily.

I appreciate the comment addresses a few of these concerns, but might be worth putting into the code too.

Reply:
good point. I made the changes you suggested. I also added a 5 minute to the overall timeout value, to reduce the risk of un-necessary timeout for files with low number of pages, in case there is some warming up time when ocrMyPdf starts processing a file. Let me know if you think this is not needed